### DISCUSSION TITLE
Can I pass in the log level as a parameter?

### DISCUSSION THREAD

**Discussion by persona94** [gabime/spdlog#2900] [Q&A]
Lets say I want to figure out the log level at run time

`void print_foo(int level, const std::string& message) {   
   spdlog:log(level, message);   
}`

How do I do this? In my code I create the spdlog::logger object at start up and use macros throughout code to print

`
#define LOG_DEBUG(format, ...) SPDLOG_DEBUG(format, ##__VA_ARGS__)

#define LOG_INFO(format, ...) SPDLOG_INFO(format, ##__VA_ARGS__)
`

I do this because I want to use my custom flags, and format the log with line number, file number and function name

**Comment by tt4g** [comment]
Use `SPDLOG_LOGGER_CALL`. This is used by each logging macro.

https://github.com/gabime/spdlog/blob/508d20f0facf33b104f42849c29149041d2324c6/include/spdlog/spdlog.h#L286-L292
